# Практическая работа №2: Обработка выборочных данных. Нахождение точечных оценок параметров распределения

Выполнили студенты гр. 2381 Соколов Сергей и Тищенко Артем. Вариант №22

## Цель работы

Получение практических навыков нахождения точечных статистических оценок параметров распределения.

## Основные теоретические положения

В практической работе будут рассмотренны следующие точечные статистические оценки:

- Начальный эмпирический момент $k$-го порядка:

  $\bar{M}_k = \frac{1}{N}\sum n_j x_j^k$;

  $N = \sum n_j - $ размер выборки; $n_j - $ абсолютная частота $j$-й варианты

- Выборочное среднее $\bar{x}_в = \bar{M}_1$

- Центральный эмпирический момент $k$-го порядка:

  $\bar{m}_k=\frac{1}{N} \sum n_j (x_j - \bar{x}_в)^k$

- Дисперсия $D = \bar{m}_2$, эта оценка является смещенной

- Исправленная оценка дисперсии: $s^2 = \frac{N}{N-1}D_в$

- Среднеквадратическое отклонение:
  - Смещенное: $σ_в = \sqrt{D_в}$
  - Исправленное: $s = \sqrt{s^2}$

- Асимметрии: $\bar{A}_s = \frac{\bar{m}_3}{s^3}$

- Эксцесса: $\bar{E} = \frac{\bar{m}_4}{s^4} - 3$

- Условные эмпирические моменты:

  $\bar{M}_k^*= \frac{1}{N} \sum n_j \left( \frac{x_j - C}{h}\right)^k$;

  $C -$ условный ноль, значения которого выбирается равным значению варианты интервального ряда, являющейся средней или близкой к средней по значению в этом ряду; $h-$ длина интервала.

- Мода, $M_o^*$ $-$ варианта с наибольшей частотой.

Для интервального ряда: $M_o^* = x_{i-1} + h\frac{\tilde{m}_i - \tilde{m}_{i-1}}{\tilde{m}_i - \tilde{m}_{i-1} + \tilde{m}_i - \tilde{m}_{i-1}}$, $i -$ номер интервала наибольшей частоты, $\tilde{m}_i - $ относительная частота ряда $i$

- Медиана, $M_e^* -$ варианта, соответствующая середине ранжированного ряда.
Для интервального ряда: $M_e^* = x_{i-1} + \frac{h}{\tilde{m}_i} \left(0.5 - \overset{i-1}{\underset{j=1}{\sum}}\tilde{m}_j  \right)$

## Постановка задачи

Для заданных выборочных данных вычислить с использованием метода моментов и условных вариант точечные статистические оценки математического ожидания, дисперсии, среднеквадратичного отклонения, асимметрии, эксцесса, моды, медианы и коэффициента вариации исследуемой случайной величины. Полученные результаты содержательно проинтерпретировать.

## Выполнение работы


Подключение библиотек и загрузка интервального ряда из ПР1

In [13]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

N = 112
df = pd.read_csv('is.csv', index_col=0)
df

,[xi; xi+1),mid_xi,mi,rel_mi,s_mi,s_rel_mi
0,[320.0; 353.875),336.9375,3,0.026786,3,0.026786
1,[353.875; 387.75),370.8125,8,0.071429,11,0.098214
2,[387.75; 421.625),404.6875,18,0.160714,29,0.258929
3,[421.625; 455.5),438.5625,28,0.250000,57,0.508929
4,[455.5; 489.375),472.4375,24,0.214286,81,0.723214
5,[489.375; 523.25),506.3125,18,0.160714,99,0.883929
6,[523.25; 557.125),540.1875,8,0.071429,107,0.955357
7,[557.125; 591.0],574.0625,5,0.044643,112,1.000000


#### Задание 1
Для середин интервального ряда, полученного в практической работе №1, вычислить условные варианты. Заполнить табл. 1 (в последней строке
Σ
 необходимо заполнить суммы столбцов; ячейки отмеченные прочерком заполнять не надо). Провести контроль вычислений.

In [14]:
df1 = df[['mid_xi', 'rel_mi']]
df1 = df1.rename(columns={'rel_mi': 'ni'})
df1.head()

,mid_xi,ni
0,336.9375,0.026786
1,370.8125,0.071429
2,404.6875,0.160714
3,438.5625,0.250000
4,472.4375,0.214286


In [15]:
C = df1['mid_xi'][len(df1) // 2]
h = df1['mid_xi'][1] - df1['mid_xi'][0]

df1.loc[:,'ui'] = ((df1.loc[:, 'mid_xi'] - C ) / h).astype(int)

for i in range(1, 5):
  df1.loc[:,f'ni*ui^{i}'] = (df1.loc[:, 'ni'] * df1.loc[:, 'ui'] ** i)
df1.loc[:,f'ni*(ui + 1)^{4}'] = (df1.loc[:, 'ni'] * (df1.loc[:, 'ui'] + 1) ** 4)

df1.loc['Σ'] = [None, df1['ni'].sum(), None, df1['ni*ui^1'].sum(), df1['ni*ui^2'].sum(), df1['ni*ui^3'].sum(), df1['ni*ui^4'].sum(), df1['ni*(ui + 1)^4'].sum()]
df1

,mid_xi,ni,ui,ni*ui^1,ni*ui^2,ni*ui^3,ni*ui^4,ni*(ui + 1)^4
0,336.9375,0.026786,-4.0,-0.107143,0.428571,-1.714286,6.857143,2.169643
1,370.8125,0.071429,-3.0,-0.214286,0.642857,-1.928571,5.785714,1.142857
2,404.6875,0.160714,-2.0,-0.321429,0.642857,-1.285714,2.571429,0.160714
3,438.5625,0.250000,-1.0,-0.250000,0.250000,-0.250000,0.250000,0.000000
4,472.4375,0.214286,0.0,0.000000,0.000000,0.000000,0.000000,0.214286
5,506.3125,0.160714,1.0,0.160714,0.160714,0.160714,0.160714,2.571429
6,540.1875,0.071429,2.0,0.142857,0.285714,0.571429,1.142857,5.785714
7,574.0625,0.044643,3.0,0.133929,0.401786,1.205357,3.616071,11.428571
Σ,NaN,1.000000,NaN,-0.455357,2.812500,-3.241071,20.383929,23.473214


Получим таблицу:

|  $i$  | $\tilde{xi}$   |   $n_i$ | $u_i$   |   $n_i$ $\cdot$ $u_i$ |   $n_i$ $\cdot $ $ u_i^2$ |   $n_i$ $\cdot$ $u_i^3$ |   $n_i$ $\cdot$ $u_i^4$ |   $ n_i$ $\cdot$ ($u_i$ + 1$)^4$ |
|:---|:---------|----------:|:-----|----------:|----------:|----------:|----------:|----------------:|
| 0  | 336.9375 | 0.0267857 | -4   | -0.107143 |  0.428571 | -1.71429  |  6.85714  |        2.16964  |
| 1  | 370.8125 | 0.0714286 | -3   | -0.214286 |  0.642857 | -1.92857  |  5.78571  |        1.14286  |
| 2  | 404.6875 | 0.160714  | -2   | -0.321429 |  0.642857 | -1.28571  |  2.57143  |        0.160714 |
| 3  | 438.5625 | 0.25      | -1   | -0.25     |  0.25     | -0.25     |  0.25     |        0        |
| 4  | 472.4375 | 0.214286  | 0    |  0        |  0        |  0        |  0        |        0.214286 |
| 5  | 506.3125 | 0.160714  | 1    |  0.160714 |  0.160714 |  0.160714 |  0.160714 |        2.57143  |
| 6  | 540.1875 | 0.0714286 | 2    |  0.142857 |  0.285714 |  0.571429 |  1.14286  |        5.78571  |
| 7  | 574.0625 | 0.0446429 | 3    |  0.133929 |  0.401786 |  1.20536  |  3.61607  |       11.4286   |
| Σ  | $-$        | 1         | $-$    | -0.455357 |  2.8125   | -3.24107  | 20.3839   |       23.4732   |

##### Контроль вычислений:
$$   \sum n_i \cdot u_i^4 + 4 * \sum n_i \cdot u_i^3 + 6 \sum n_i \cdot   u_i^2 + \sum n_i \cdot u_i + 1 - \sum  n_i \cdot (u_i + 1)^4 = 0$$

In [16]:
df1.iloc[-1, -2] + 4 * df1.iloc[-1, -3] + 6 * df1.iloc[-1, -4] + 4 * df1.iloc[-1, -5] + 1 - df1.iloc[-1, -1]

np.float64(0.0)

При контроле вычислений был получен $0$, следовательно значения были посчитаны верно.

#### Задание 2
Вычислить условные эмпирические моменты $\bar{M^*_k}$
 через условные варианты. С помощью условных эмпирических моментов вычислить центральные эмпирические моменты
$\bar{m_k}$
. Полученные результаты занести в табл. 2.

In [17]:
df2 = pd.DataFrame(index=range(1, 5), columns=['M^*k', 'mk'])
for i in range(1, 5):
  df2.loc[i, 'M^*k'] = df1.iloc[-1, 2 + i]


M1 = df2.loc[1, 'M^*k']
M2 = df2.loc[2, 'M^*k']
M3 = df2.loc[3, 'M^*k']
M4 = df2.loc[4, 'M^*k']
df2.loc[1, 'mk'] = 0
df2.loc[2, 'mk'] = (M2 - M1 ** 2) * (h ** 2)
df2.loc[3, 'mk'] = (M3 - 3 * M2 * M1 + 2 * M1 ** 3) * (h ** 3)
df2.loc[4, 'mk'] = (M4 - 4 * M3 * M1 + 6 * M2 * M1 ** 2 - 3 * M1 ** 4) * (h ** 4)
df2

,M^*k,mk
1,-0.455357,0
2,2.8125,2989.450184
3,-3.241071,16021.822263
4,20.383929,23505528.765329


Получим таблицу:

|    |$\bar{M}^*_k$|   $\bar{m}_k$ |
|---:|----------:|----------------:|
|  1 | -0.455357 |     0           |
|  2 |  2.8125   |  2989.45        |
|  3 | -3.24107  | 16021.8         |
|  4 | 20.3839   |     2.35055e+07 |

#### Задание 3
Вычислить выборочные среднее
$\bar{x}_в$
 и дисперсию
$D_в$
 с помощью стандартной формулы и с помощью условных вариант. Убедиться, что результаты совпадают. Вычислить выборочное СКО $σ_в$

##### Выборочное среднее $\bar{x}_в$

In [18]:
# Через условные варианты
xv1 = df2.iloc[0,0] * h + C

# Через стандартную формулу
xv2 = df1.loc[:7, 'mid_xi'].to_numpy().dot(df1.loc[:7, 'ni'])
print(f"Выборочные среднее через условные варианты: {xv1} \nВыборочные среднее через стандартные формулы: {xv2} \nРазница: {xv1 - xv2}")

Выборочные среднее через условные варианты: 457.0122767857143 
Выборочные среднее через стандартные формулы: 457.0122767857141 
Разница: 1.7053025658242404e-13


Видно, что оба числа практически совпадают. Разница могла возникнуть в следствии ошибок округления при рассчетах.

Выборочная дисперсия $D_в$

In [19]:
# Через условные варианты
D1 = df2.loc[2, 'mk']

# Через стандартную формулу
D2 = ((df1.loc[:7, 'mid_xi'] - xv2) ** 2).to_numpy().dot(df1.loc[:7, 'ni'].to_numpy())
print(f"Дисперсия через условные варианты: {D1} \nДисперсия через стандартные формулы: {D2} \nРазница: {D1 - D2}")

Дисперсия через условные варианты: 2989.4501841019587 
Дисперсия через стандартные формулы: 2989.4501841019587 
Разница: 0.0


Оба значения совпали

##### Выборочное СКО $σ_в$

In [20]:
print(np.sqrt(D1))

54.675864731176944


#### Задание 4
Вычислить исправленную выборочную дисперсию $s^2$ и исправленное СКО $s$. Сравнить данные оценки с смещёнными оценками дисперсии и СКО. Сделать выводы.

In [21]:
# Исправленная дисперсия
s2 = N * D1 / (N - 1)
# Исправленное СКО
s = np.sqrt(s2)

print(f"Исправленная дисперсия: {s2} \nИсправленное СКО: {s} \nРазница: \nДисперсий: {s2 - D1} \n СКО: {s - np.sqrt(D1)}")

Исправленная дисперсия: 3016.3821677425167 
Исправленное СКО: 54.92160019284322 
Разница: 
Дисперсий: 26.931983640557974 
 СКО: 0.24573546166627835


Исправленные оценки оказались немного больше смещенных оценок в виду достаточно большого размера выборки $N=112$.

#### Задание 5
Найти статистическую оценку коэффициентов асимметрии $\bar{A}_s$ и эксцесса $\bar{E}$. Сделать выводы.

In [22]:
As = df2.loc[3, 'mk'] / (s ** 3)
E = df2.loc[4, 'mk'] / (s ** 4) - 3
print(f"Статистическая оценка коэффициента асимметрии: {As} \nСтатистическая оценка эксцесса: {E}")

Статистическая оценка коэффициента асимметрии: 0.09671244494185104 
Статистическая оценка эксцесса: -0.4165664461907408


Коэффициент асимметрии больше нуля, следовательно большинство элемнтов выборки находится слево от среднего значения; мода больше медианы.

Коэффициент эксцесса меньше нуля, следовательно пик более плоский

#### Задание 6
Для интервального ряда вычислить моду $M^*_o$, медиану $M^*_e$ и коэффициент вариации $V^*$ заданного распределения. Сделать выводы.

In [23]:
Mo = 421.625 + h * (df.loc[3, 'rel_mi'] - df.loc[2, 'rel_mi']) / ((df.loc[3, 'rel_mi'] - df.loc[2, 'rel_mi']) + df.loc[3, 'rel_mi'] - df.loc[4, 'rel_mi'])
Me = 421.625 + h / df.loc[3, 'rel_mi'] * (0.5 - (df.loc[:2, 'rel_mi']).sum())
V = np.sqrt(D2) / xv2 * 100

print(f"Мода: {Mo} \nМедиана: {Me} \nКоэффициент вариации: {V}%")

Мода: 445.82142857142856 
Медиана: 454.2901785714286 
Коэффициент вариации: 11.963762793360058%


Мода больше медианы, что подтверждает положительное значение коэффикиента асимметрии.

Коэффициент вариации показывает, что значения варианты  не сильно разбросаны относительно $\bar{x}_в$

## Выводы

Были получены практические навыки нахождения точечных статистических оценок параметров распределения. Для выборки были посчитаны точечные статистичкескии оценки условных эмпирических моментов, центральных эмпирических моментов, выборочного среднего и дисперсии с использованием условных вариант и стандартных формул, используя оба способа были получены одинаковые значения, СКО, исправленной дисперсии, исправленную СКО, коэффициента асимметрии, коэффициента эксцесса, моды, медианы, коэффициента вариации.